frontend/src/components/profileSetting/ProfileSetting.jsx

```diff
@@ -2,16 +2,18 @@ import { Avatar, Button, TextField } from "@mui/material";
 import "./profileSetting.css";
 import useFetch from "../../useFetch";
 import axios from "axios";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { Navigate, useNavigate } from "react-router-dom";
-
+import useGetAuthorID from "../../useGetAuthorID.js"
+import useGetAuthorData from "../../useGetAuthorData";
 
 
 
 export default function ProfileSetting() {
+    
+    // grabbing data from /get_author_id/ and /authors/author_id APIs
+    const {data, loading1, error1, authorID} = useGetAuthorData("/get_author_id/");
```

Comment:
No need to pass the url here since that is constant. Use it directly inside the function.

Reply:
I think it would be better if you split `useGetAuthorData()` into `useGetAuthorID()` and `useGetAuthorData()`. Cause they might be places we need to get the current author Id to check if user is authenticated.

Reply:
> No need to pass the url here since that is constant. Use it directly inside the function.

Yeah I had this beforehand but then I thought that if any change were to happen in the get_author_api then I thought it'll be easier to change in the components themselves but thinking about it now, it might be best to edit once in the function itself so I'll do that once I touchdown.

Reply:
> I think it would be better if you split `useGetAuthorData()` into `useGetAuthorID()` and `useGetAuthorData()`. Cause they might be places we need to get the current author Id to check if user is authenticated.

I actually created useGetAuthorID() last night but calling both useGetAuthorID() and useGetAuthorData() in the same function caused many issues that made me stuck for hours. But I'll put back useGetAuthorID() later regardless. Hopefully we can find a way to call two different get calls under 1 component! Any solution found would be much appreciated!